myia/pipeline/steps.py

```diff
@@ -614,11 +618,17 @@ def convert_arg(self, arg, orig_t: AbstractScalar, backend):
     elif issubclass(t, dtype.Nil):
         if arg is not None:
             raise MyiaInputTypeError(f'Expected None')
+    elif issubclass(t, dtype.String):
+        if not isinstance(arg, str):
+            raise MyiaInputTypeError(f'Expected string')
+        arg = str_to_tag(arg)
+        t = dtype.Int[64]
     else:
         raise MyiaInputTypeError(f'Invalid type: {t}')
-    expected_value = orig_t.values[VALUE]
-    if expected_value is not ANYTHING and expected_value != arg:
-        raise MyiaInputTypeError(f'Invalid value: {arg}')
+    if not issubclass(orig_t.values[TYPE], dtype.String):
```

Comment:
Why is that check here? If the program is compiled assuming that we are giving the string "hello" as a parameter, then we should check that this is the right string.